pkg/gofr/datasource/file/ftp/fs.go

```diff
@@ -0,0 +1,276 @@
+	Password  string // FTP password
+	Port      int    // FTP port
+	RemoteDir string // Remote directory path. Base Path for all FTP Operations.
+}
+
+// New initializes a new instance of ftpFileSystem with provided configuration.
+func New(config *Config) FileSystem {
+	return &ftpFileSystem{config: config}
+}
+
+// UseLogger sets the logger interface for the FTP file system.
+func (f *ftpFileSystem) UseLogger(logger interface{}) {
+	if l, ok := logger.(Logger); ok {
+		f.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the ftpFileSystem client which asserts the Metrics interface.
+// Currently not implemented.
+func (*ftpFileSystem) UseMetrics(_ interface{}) {
+
+}
+
+// Connect establishes a connection to the FTP server and logs in.
+func (f *ftpFileSystem) Connect() {
+	ftpServer := fmt.Sprintf("%v:%v", f.config.Host, f.config.Port)
+
+	const dialTimeout = 5 * time.Second
+
+	conn, err := ftp.Dial(ftpServer, ftp.DialWithTimeout(dialTimeout))
+	if err != nil {
+		f.logger.Errorf("Failed to connect to FTP server: %v", err)
+		return
+	}
+
+	f.conn = &Conn{conn}
+
+	err = conn.Login(f.config.User, f.config.Password)
+	if err != nil {
+		f.logger.Errorf("Failed to login: %v", err)
+	}
+
+	f.logger.Logf("Login Successful")
+}
+
+// Create creates an empty file on the FTP server.
+func (f *ftpFileSystem) Create(name string) (File, error) {
+	if name == "" {
+		return nil, errEmptyFilename
+	}
+
+	emptyReader := new(bytes.Buffer)
+
+	filePath := fmt.Sprintf("%s/%s", f.config.RemoteDir, name)
+
+	err := f.conn.Stor(filePath, emptyReader)
+	if err != nil {
+		return nil, err
+	}
+
+	f.logger.Logf("Created %s", name)
+
+	s := strings.Split(filePath, "/")
+
+	res, err := f.conn.Retr(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Close()
+
+	return &ftpFile{
+		response: res,
+		name:     s[len(s)-1],
+		path:     filePath,
+		conn:     f.conn,
+		logger:   f.logger,
+	}, nil
+}
+
+// Mkdir creates a directory on the FTP server. Here, os.FileMode is unused.
+func (f *ftpFileSystem) Mkdir(name string, _ os.FileMode) error {
+	if name == "" {
+		return errEmptyDirectory
+	}
+
+	filePath := fmt.Sprintf("%s/%s", f.config.RemoteDir, name)
+
+	err := f.conn.MakeDir(filePath)
+	if err == nil {
+		f.logger.Logf("Created %s directory", name)
+		return nil
+	}
+
+	return err
+}
+
+// MkdirAll creates directories recursively on the FTP server. Here, os.FileMode is unused.
+func (f *ftpFileSystem) MkdirAll(path string, _ os.FileMode) error {
+	dirs := strings.Split(path, "/")
+
+	currentDir := dirs[0]
+
+	v := 0
+
+	for i, dir := range dirs {
+		// Ignore empty directory names (can happen if there are double slashes).
+		if dir == "" {
+			continue
+		}
+
+		if i == 0 {
+			currentDir = dir
+		} else {
+			currentDir = fmt.Sprintf("%s/%s", currentDir, dir)
+		}
+
+		err := f.conn.MakeDir(currentDir)
+		if err != nil {
+			// if error indicates that directory exists continue, else return.
+			if fmt.Sprint(err) != directoryAlreadyExistError {
```

Comment:
[bug] Comment and log disagree with if conditional. In this case we would continue when the error is _not_ `directoryAlreadyExistError`

[fragile test] What happens if the server returns `"550 Fehler bei der Erstellung des Verzeichnis."` because the server is running with `LANG=de`?

Reply:
This issue will be handled in the next iterations of improvement of this feature.